interfaces/policy/basedeclaration_test.go

```diff
@@ -1003,3 +1005,87 @@ plugs:
 		checkOpticalDriveAutoConnect(plugYaml, NotNil)
 	}
 }
+
+func (s *baseDeclSuite) TestRawVolumeOverride(c *C) {
+	slotYaml := `name: slot-snap
+type: gadget
+version: 0
+slots:
+  raw-volume:
+    path: /dev/mmcblk0p1
+`
+	slotSnap := snaptest.MockInfo(c, slotYaml, nil)
+	// mock a well-formed slot snap decl with SnapID
+	slotSnapDecl := s.mockSnapDecl(c, "slot-snap", "slotsnapidididididididididididid", "canonical", "")
+
+	plugYaml := `name: plug-snap
+version: 0
+plugs:
+  raw-volume:
+`
+	plugSnap := snaptest.MockInfo(c, plugYaml, nil)
+
+	// no plug-side declaration
+	cand := &policy.ConnectCandidate{
+		Plug:                interfaces.NewConnectedPlug(plugSnap.Plugs["raw-volume"], nil, nil),
+		Slot:                interfaces.NewConnectedSlot(slotSnap.Slots["raw-volume"], nil, nil),
+		SlotSnapDeclaration: slotSnapDecl,
+		BaseDeclaration:     s.baseDecl,
+	}
+
+	err := cand.Check()
+	c.Check(err, NotNil)
+	c.Assert(err, ErrorMatches, "connection denied by slot rule of interface \"raw-volume\"")
+	err = cand.CheckAutoConnect()
```

Comment:
nowadays this returns the arity as well,  probably need to merge master

Reply:
Done